### PR TITLE
[WIP]reform for extends

### PR DESCRIPTION
**Pull request by leizhiyuan**
### Issues associated with this PR

reform tracer , make more powerful for extend. improve performace.

but if we need more span field like router,other time ,we need to add to open source code.

### Sign the CLA
Make sure you have signed the [CLA](https://www.clahub.com/agreements/alipay/sofa-mosn)

### Solutions
You should show your solutions about the issues in your PR, including the overall solutions, 
details and the changes. At this time, Chinese is allowed to describe these.

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases.
And you need to show the ut result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

File: pkg/trace/sofatracer.go [hunk 1]
```diff
@@ -152,71 +138,125 @@ func (tracer *SofaTracer) Start(startTime time.Time) types.Span {
...
+			// The time server(upstream) takes to process the RPC
+			// server.duration = server.pool.wait.time + biz.impl.time + resp.serialize.time + req.deserialize.time
+			duration := strconv.FormatInt(span.endTime.Sub(span.startTime).Nanoseconds()/1000000, 10)
+
+			printData.WriteString("\"server.duration\":")
+			printData.WriteString("\"" + duration + "\"")
+			printData.WriteString("}")
+			printData.WriteString("\n")
+
+			return tracer.egressLogger.Print(printData, true)
```
Comment (taoyuanyuan) [inline pkg/trace/sofatracer.go]: should be tracer.ingressLogger.Print
